### PR TITLE
WIP: apis: sketch apis.kcp.dev future extensions

### PR DESCRIPTION
**Pull request by sttts**
Needs https://github.com/kcp-dev/kcp/pull/583.

### CODE REVIEW

File: pkg/apis/exports/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+// +kubebuilder:printcolumn:name="Workspace",type="string",JSONPath=`.reference.workspace`,priority=2
+// +kubebuilder:printcolumn:name="Name",type="string",JSONPath=`.reference.name`,priority=3
+type Export struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state.
+	// +required
+	Reference ExportReference `json:"reference"`
```
Comment (njhale) [inline pkg/apis/exports/v1alpha1/types.go:61]: nit: The go doc comment doesn't match the field name.

File: pkg/admission/apiresourceschema/admission.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	"fmt"
+	"io"
+
+	"k8s.io/apiserver/pkg/admission"
+
+	kcpadmissionhelpers "github.com/kcp-dev/kcp/pkg/admission/helpers"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+)
+
+// Validate ClusterWorkspace creation and updates for valid use of spec.type, i.e. the
```
Comment (njhale) [inline pkg/admission/apiresourceschema/admission.go:30]: nit: vestigial copy-pasta comment

File: pkg/apis/exports/v1alpha1/types.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	// kubebuilder:validation:MinLength=1
+	// +required
+	Resource string `json:"resource"`
+
+	// workspace is the workspace the referenced object is in.
+	//
+	// If the workspace is empty, the referenced object is in the same workspace.
+	//
+	// kubebuilder:validation:MinLength=1
+	Workspace string `json:"workspace"`
```
Comment (sttts) [inline pkg/apis/exports/v1alpha1/types.go:83]: Today we would probably make the export discoverable for users and just put the reference key `base62(sha256(<workspace>:<name>))` here.

**Comment by ncdc** [comment]
As discussed on Slack, closing for now, but it'll still be in the repo so we can come back and look at it again for ideas if needed.

**Pull request closed without merging**